File: src/dao/model/Portfolio.ts
interface Portfolio {
  id: number; // Hash Key
  symbol: string; // Sory Key
  quantity: number;
  avgCostBasis: number; // Retain 8 decimal places
  lock: number; // Unix Timestamp
  createdAt: number; // Unix Timestamp
  updatedAt: number; // Unix Timestamp
}

export type { Portfolio };

File: src/model/portfolio.py
from pydantic import BaseModel
from typing import Optional


class Portfolio(BaseModel):
    id: int  # Hash Key
    symbol: str  # Sort Key
    quantity: int  # Retain 4 decimal places
    avgCostBasis: int  # Retain 8 decimal places
    lock: Optional[int] = None  # Unix Timestamp
    createdAt: int  # Unix Timestamp
    updatedAt: int  # Unix Timestamp

    class Config:
        from_attributes = True
